src/firefly/js/visualize/ui/CatalogTableListField.jsx

```diff
@@ -0,0 +1,227 @@
+        //    fields: FieldGroupUtils.getGroupFields(this.props.groupKey)
+        //};
+    }
+
+    //
+    //componentWillUnmount() {
+    //    if (this.removeListener) this.removeListener();
+    //    this.iAmMounted = false;
+    //}
+    //
+    //componentDidMount() {
+    //    this.iAmMounted = true;
+    //    this.removeListener = FieldGroupUtils.bindToStore(this.props.groupKey, (fields) => {
+    //        if (this.iAmMounted) this.setState({fields});
+    //    });
+    //}
+
+    /**
+     * The function should be used to fetch the DD from a catalog named passed in as argument catName
+     * and used to populate a table with column definition, description and any DD fetched
+     * @param {string} catName the catalog name to get DD fetched from
+     */
+    updateTable(catName = 'wise_allwise_p3as_psd') {
+        this.getDD(catName);//'wise_allwise_p3as_psd'
+    }
+
+    /**
+     * Getting dd info from catalog name catName
+     * @param catName string name of the catalog for searching DD information
+     */
+    getDD(catName) {
+
+
+        const request = {id: 'GatorDD', 'catalog': catName}; //Fetch DD master table
+        TblUtil.doFetchTable(request).then((tableModel) => {
+
+            var data = tableModel.tableData.data;
+            const html = data.map((c) => {
+                return c[0] + ':' + c[1] + ' [' + c[2] + ']';
+            });
+            const cols = tableModel.tableData.columns;
+            // What to do with that?
+            // this.setState({...state},{})
+
+        }).catch((reason) => {
+                console.error(reason);
+            }
+        );
+    }
+
+    componentWillReceiveProps(np) {
+        const newFirstCatalogValue = get(np, 'data[0].value', '');
+        const oldFirstValueCatalog = get(this.props, 'data[0].value', '');
+        if (newFirstCatalogValue != oldFirstValueCatalog) {
+            this.setState(...this.state, {idx: 0});
+        } else {
+            this.setState(...this.state, {idx: np.indexClicked});
+        }
+
+    }
+
+    /**
+     * Only update when new catalog selected or different list item has been clicked
+     * @param np
+     * @returns {boolean}
+     */
+    shouldComponentUpdate(np) {
+        const newCatValue = get(np, 'value', '');
+        const oldCatValue = get(this.props, 'value', '');
+        //FIXME Doesn't work when user comes back from another tab
+        return newCatValue != oldCatValue;
+    }
+
+    render() {
+        var {data, cols, onClick, fieldKey} = this.props;//data {cat:...}
+        const indexClicked = get(this.state, 'idx', 0);
+
+        if (isEmpty(data)) {
+            return (
+                <div style={{position: 'relative'}}>
+                    <div className='loading-mask'/>
+                </div>
+            );
+        }
+
+        var items = data.map((o, index) => {
+
+
+            let clicked = false;
+            if (indexClicked === index) {
+                clicked = true;
+            }
```

Comment:
I like that!

On 5/16/16 3:53 PM, loitly wrote:
>
> In src/firefly/js/visualize/ui/CatalogTableListField.jsx 
> <https://github.com/Caltech-IPAC/firefly/pull/85#discussion_r63439629>:
>
> > +
> > +        if (isEmpty(data)) {
> > +            return (
> > +                <div style={{position: 'relative'}}>
> > +                    <div className='loading-mask'/>
> > +                </div>
> > +            );
> > +        }
> > +
> > +        var items = data.map((o, index) => {
> > +
> > +
> > +            let clicked = false;
> > +            if (indexClicked === index) {
> > +                clicked = true;
> > +            }
>
> isn't this same as
> const clicked = indexClicked === index;
> you can actually remove the whole block and put that condition into 
> the bottom attribute.. ie.
> isClicked={indexClicked === index}
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/Caltech-IPAC/firefly/pull/85/files/8bdfd737fd8631b45d10aff1ebb2742bee675e9e#r63439629>
>

-- 
Emmanuel Joliet
NASA/IPAC  Infrared Science Archive
MS 100-22, Caltech, Pasadena, CA 91125
Office: 170
Email:  ejoliet@ipac.caltech.edu
Phone:  (626) 395-1489